### PR TITLE
fix: ignore programmatic clicks when disabled

### PR DESCRIPTION
**Pull request by TatuLund**
fixes https://github.com/vaadin/flow-components/issues/5414

### CODE REVIEW

**Review by vursen — CHANGES_REQUESTED** [review]
Could you please reference a ticket? It's required for review.

**Comment by TatuLund** [comment]
@vursen I created the ticket and linked it.

**Review by vursen — COMMENTED** [review]
I'd argue against throwing exceptions and suggest just ignoring such clicks. This would align with the native browser behavior which ignores programmatic clicks on disabled buttons without throwing any exceptions:

```html
<button disabled>Click!</button>
<script>
const button = document.querySelector('button');
button.addEventListener('click', () => {
  console.log('click');
});
button.click();
</script>
```

**Comment by yuriy-fix** [comment]
@TatuLund, is it Ok if we adjust the PR to avoid throwing an exception?

**Comment by TatuLund** [comment]
@yuriy-fix basically yes, but what is your idea for change?

**Comment by yuriy-fix** [comment]
Following https://github.com/vaadin/flow-components/pull/5392#pullrequestreview-1630981248 comment, ignore  programmatic clicks on disabled buttons without throwing exceptions.

File: vaadin-button-flow-parent/vaadin-button-flow/src/test/java/com/vaadin/flow/component/button/tests/ButtonTest.java [hunk 1]
```diff
@@ -300,29 +314,19 @@ public void disableOnClick_click_disablesComponent() {
...
                 changeTracker.get("disabled"));
-        Assert.assertFalse("Button should be disabled", button.isEnabled());
-
-        changeTracker.clear();
-
-        button.addClickListener(event -> event.getSource().setEnabled(true));
+    }
 
+    @Test
+    public void disableOnClick_clickRevertsDisabled_componentIsEnabled() {
```
Comment (vursen) [inline vaadin-button-flow-parent/vaadin-button-flow/src/test/java/com/vaadin/flow/component/button/tests/ButtonTest.java:326]: **note:** The previous version of the test attempted to click the disabled button and that expectedly stopped working after the PR change.

File: vaadin-button-flow-parent/vaadin-button-flow/src/main/java/com/vaadin/flow/component/button/Button.java [hunk 2]
```diff
@@ -271,15 +271,17 @@ public boolean isIconAfterText() {
...
+     * side, but does not execute possible client side registered listeners.
      *
      * @see #clickInClient()
      */
     public void click() {
-        fireEvent(new ClickEvent<>(this, false, 0, 0, 0, 0, 0, 0, false, false,
-                false, false));
+        if (isEnabled()) {
+            fireEvent(new ClickEvent<>(this, false, 0, 0, 0, 0, 0, 0, false,
+                    false, false, false));
```
Comment (vursen) [inline vaadin-button-flow-parent/vaadin-button-flow/src/main/java/com/vaadin/flow/component/button/Button.java:283]: **note:** It seems that you can still get click triggered on a disabled button by manually firing this click event.

**Review by vursen — APPROVED** [review]

**Pull request merged**